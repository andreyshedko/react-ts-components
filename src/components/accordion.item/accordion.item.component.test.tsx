import React from "react";

import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { AccordionItemComponent } from "./accordion.item.component";

configure({ adapter: new Adapter() });

describe("Accordion Item Component", () => {
    test("should display component", async () => {
        const component = renderer.create(
            <AccordionItemComponent
                config={{
                    title: 'Title',
                    body: {
                        title: 'Sub Title',
                        element: 'element'
                    },
                    isOpened: false
                }}
                isOpened={false}
                onClick={jest.fn()}
                onKeyDown={jest.fn()}
            />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});