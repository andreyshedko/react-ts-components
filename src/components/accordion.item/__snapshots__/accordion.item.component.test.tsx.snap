// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Item Component should display component 1`] = `
<main
  className="sc-AxjAm cwuoKc"
  tabIndex={0}
>
  <section
    className="sc-AxirZ gjKCvg"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="sc-AxhCb kLnTaK"
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-down fa-w-14 "
        data-icon="chevron-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <h3
      className="sc-AxiKw ibCGdu"
    >
      Title
    </h3>
  </section>
  <section
    className="sc-AxhUy fJxdGm"
  >
    <h4>
      Sub Title
    </h4>
    <div
      className="sc-AxgMl cwaAQQ"
    >
      element
    </div>
  </section>
</main>
`;

exports[`Accordion Item Component should display items 1`] = `
<main
  className="sc-AxjAm cwuoKc"
  tabIndex={0}
>
  <section
    className="sc-AxirZ gjKCvg"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="sc-AxhCb kLnTaK"
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-down fa-w-14 "
        data-icon="chevron-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
    <h3
      className="sc-AxiKw ibCGdu"
    >
      Title
    </h3>
  </section>
  <section
    className="sc-AxhUy fJxdGm"
  >
    <h4>
      Sub Title
    </h4>
    <div
      className="sc-AxgMl cwaAQQ"
    >
      element
    </div>
  </section>
</main>
`;
